### DISCUSSION TITLE
Accessing the graphical interface for GRBL settings in IOsender.

### DISCUSSION THREAD

**Discussion by IanHarlow** [terjeio/ioSender#151] [Q&A] [answered]
Looking at the documentation and talking with some friends, I noticed that I am seeing a completely different GRBL settings tab than everyone else. Mine looks closer to a spreadsheet than a dropdown menu. I am wondering how I can access or set up the version of IOsender with the drop down menu for settings.
![lol](https://user-images.githubusercontent.com/30778313/141244608-c5fbf50e-11e4-4e46-863d-d66ddb3b4453.PNG)

I attached a screenshot to show you what I mean.

Any suggestions/help would be greatly appreciated. Thank you in advance!

Edit: I am trying to get it to look more like this (https://github.com/terjeio/ioSender/blob/master/Media/Sender3.png)

**Comment by terjeio** [comment, marked as answer]
Which controller do you have? Looks like it could be grblHAL and if so an old version?
BTW it is a bug that no field names and descriptions are shown...

The treeview (or dropdown menu) is generated from [data downloaded](https://github.com/grblHAL/core/wiki/Report-extensions#code-sets-and-enumerations) from the controller, support for that was added to grblHAL earlier this year. It solved a long standing problem for me to keep the sender UI in sync with the settings available from the controller.

**Reply by IanHarlow** [reply]
I am using Phil Barrett's Teensy 4.1 grblHAL board (T41U5XBB). I am using the latest example firmware that he provides on his github page here (https://github.com/phil-barrett/grblHAL-teensy-4.x/blob/master/configurations.zip).

Would you recommend that I compile my own firmware with the latest version of grblHAL?

Is it possible the old grblHAL version is also the reason I am having a hard time getting my machine to home? (I have enabled $22, $43,$44,$45 etc). Pressing the home button and typing $H do nothing (I am still able to manually jog my steppers). Every once in a while, it will throw an error:3 unrecognized command in response to typing $H in MDI.

**Reply by terjeio** [reply]
That zip was updated 16 months ago...

> Would you recommend that I compile my own firmware with the latest version of grblHAL?

Newer versions are [found here](https://github.com/phil-barrett/grblHAL-teensy-4.x/commit/bc9ed577c8874e494dcc576aedd030118b86527f) if you have a Teensy 4.1, compiling yourself is also an option and will give you freedom to configure the build as you want.

> Is it possible the old grblHAL version is also the reason I am having a hard time getting my machine to home?

Could be - update to find out. IIRC I had bugs in my USB code and there were bugs in the Arduino framework as well that long ago.

**Reply by IanHarlow** [reply]
You were 100% correct. updating to the latest firmware fixed all of my issues. Thank you so much. 

I am now able to successfully home my machine and see the dropdown settings menu.

**Comment by phil-barrett** [comment]
More recent [versions are here](https://github.com/phil-barrett/grblHAL-teensy-4.x/blob/master/hex_files/grblHAL_T41.zip).

**Reply by IanHarlow** [reply]
Thank you for your quick reply and excellent work on the board!! Updating to the firmware you linked fixed all of my problems. Now to setting up spindle control :). Thank you again
